File: urbanairship-sdk/src/test/java/com/urbanairship/push/TagUtilsTest.java
/* Copyright 2017 Urban Airship and Contributors */

package com.urbanairship.push;

import com.urbanairship.BaseTestCase;
import com.urbanairship.json.JsonException;
import com.urbanairship.json.JsonValue;

import org.junit.Test;

import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertNull;
import static junit.framework.Assert.assertTrue;

public class TagUtilsTest extends BaseTestCase {

    /**
     * Test converting a JsonValue to a tags map
     */
    @Test
    public void testConvertToTagsMap() throws JsonException {
        Map<String, Set<String>> tagGroups = new HashMap<>();
        Set<String> tags = new HashSet<>();
        tags.add("tag1");
        tags.add("tag2");
        tags.add("tag3");
        tagGroups.put("tagGroup", tags);

        JsonValue jsonValue = JsonValue.wrap(tagGroups);

        Map<String, Set<String>> map = TagUtils.convertToTagsMap(jsonValue);
        assertEquals("Map size mismatch", map.size(), tagGroups.size());
        assertTrue("Value mismatch", map.containsValue(tags));
    }

    /**
     * Test converting a null and non-JsonMap returns null.
     */
    @Test
    public void testConvertNullToTagsMap() throws JsonException {
        assertNull(TagUtils.convertToTagsMap(null));

        JsonValue jsonString = JsonValue.parseString("non-JsonMap");
        assertNull(TagUtils.convertToTagsMap(jsonString));
    }
}

File: urbanairship-core/src/test/java/com/urbanairship/channel/TagUtilsTest.kt
/* Copyright Airship and Contributors */
package com.urbanairship.channel

import androidx.test.ext.junit.runners.AndroidJUnit4
import com.urbanairship.channel.TagUtils.convertToTagsMap
import com.urbanairship.json.JsonValue
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
import org.junit.Assert.assertTrue
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
public class TagUtilsTest {

    /**
     * Test converting a JsonValue to a tags map
     */
    @Test
    public fun testConvertToTagsMap() {
        val tags = setOf("tag1", "tag2", "tag3")
        val tagGroups = mapOf(
            "tagGroup" to tags
        )

        val jsonValue = JsonValue.wrap(tagGroups)

        val map = convertToTagsMap(jsonValue)
        assertEquals("Map size mismatch", map?.size, tagGroups.size)
        assertTrue("Value mismatch", map?.containsValue(tags) == true)
    }

    /**
     * Test converting a null and non-JsonMap returns null.
     */
    @Test
    public fun testConvertNullToTagsMap() {
        assertNull(convertToTagsMap(null))

        val jsonString = JsonValue.parseString("non-JsonMap")
        assertNull(convertToTagsMap(jsonString))
    }
}
